fontc/src/lib.rs

```diff
@@ -2028,4 +2053,211 @@ mod tests {
+                            coords.end_coord().to_f32(),
+                        ]
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .collect::<Vec<_>>();
+        // Glyph "A" defines three masters (Regular, Medium and Bold) so uses two
+        // regions (the first two in the list); the rest of the glyphs only have
+        // two masters so use only one region (the last one).
+        assert_eq!(
+            region_coords,
+            vec![
+                vec![[0.0, 0.333313, 1.0]],
+                vec![[0.333313, 1.0, 1.0]],
+                vec![[0.0, 1.0, 1.0]],
+            ]
+        );
+        // in this particular test font, despite being multi-model, we still end up with
+        // a single VarData subtable, filled with zeros for regions that don't apply
+        assert_eq!(varstore.item_variation_data_count(), 1);
+        let vardata = varstore.item_variation_data().get(0).unwrap().unwrap();
+        assert_eq!(vardata.region_indexes(), &[0, 1, 2]);
+        // 4 rows insted of 5 because gid1 and gid4 happen to share the same varidx
+        assert_eq!(vardata.item_count(), 4);
+        let Some(Ok(varidx_map)) = hvar.advance_width_mapping() else {
+            panic!("Expected advance width mapping");
+        };
+        assert_eq!(
+            (0..num_glyphs)
+                .map(|gid| {
+                    let varidx = varidx_map.get(gid as u32).unwrap();
+                    assert_eq!(varidx.outer, 0);
+                    varidx.inner
+                })
+                .collect::<Vec<_>>(),
+            vec![0, 2, 3, 1, 2],
+        );
+        // .notdef is not variable
+        assert_eq!(
+            hvar.advance_width_delta(GlyphId::new(0), &[F2Dot14::from_f32(1.0)])
+                .unwrap(),
+            Fixed::from_f64(0.0)
+        );
+        // 'space' has two masters, with Bold 100 units wider than Regular
+        assert_eq!(
+            hvar.advance_width_delta(GlyphId::new(1), &[F2Dot14::from_f32(1.0)])
+                .unwrap(),
+            Fixed::from_f64(100.0)
+        );
+        // ... so it will be exactly 50 units wider half-way in between
+        assert_eq!(
+            hvar.advance_width_delta(GlyphId::new(1), &[F2Dot14::from_f32(0.5)])
+                .unwrap(),
+            Fixed::from_f64(50.0)
+        );
+        // 'A' is 120 units wider than Regular in the Bold master
+        assert_eq!(
+            hvar.advance_width_delta(GlyphId::new(2), &[F2Dot14::from_f32(1.0)])
+                .unwrap(),
+            Fixed::from_f64(120.0)
+        );
+        // ... but also has an additional Medium (wght=500, normalized 0.3333) master with
+        // a delta of 70; so at normalized 0.5 (wght=550), its delta will *not* be 60
+        assert_eq!(
+            hvar.advance_width_delta(GlyphId::new(2), &[F2Dot14::from_f32(0.5)])
+                .unwrap(),
+            Fixed::from_f64(83.0)
+        );
```

Comment:
Optional, I found this hard to read due to types (F2Dot14 didn't scream normalized coord at me), the name is clearer than the gid, and the interesting part is a bit obscured by things like Fixed coversion, building arrays of F2Dot14, etc. For me at least it would be more readable if some of the noise went away, e.g.:

```rust
// glyph_order can be obtained from the result of the compile call

assert_hvar_advance(glyph_order, vec![
 // .notdef has no variation
  (".notdef", NormalizedCoord::new(0.5), 0.0),
  (".notdef", NormalizedCoord::new(1.0), 0.0),
  // 'space' has two masters, with Bold 100 units wider than Regular
  ("space", NormalizedCoord::new(1.0), 100.0),
   // ... so it will be exactly 50 units wider half-way in between
  ("space", NormalizedCoord::new(0.5), 50.0),
]
```

Reply:
Done in https://github.com/googlefonts/fontc/pull/490/commits/7bc827faf2a0f406d57aeaaa8a8b92ba90147599